docs/en/cookbook/working-with-datetime.rst

```diff
@@ -58,25 +58,88 @@ Handling different Timezones with the DateTime Type
 
-The solution for this dilemma is simple. Don't use timezones with DateTime and Doctrine 2. However there is a workaround
-that even allows correct date-time handling with timezones:
+The solution for this dilemma seems simple. Don't use timezones with DateTime and Doctrine 2. However there are 2 workarounds
+that even allows correct date-time handling with timezones depending on your use case:
 
-1. Always convert any DateTime instance to UTC.
-2. Only set Timezones for displaying purposes
+1a. Don't convert DateTimes to UTC
+1b. Always convert any DateTime instance to UTC
+2. Set Timezones for displaying purposes
 3. Save the Timezone in the Entity for persistence.
 
+Wait: **What? Shall we convert to UTC or not?**
+
+That depends on what kind of DateTimes you are handling! When you are handling current dates like log-entries or 
+datetimes that are within one or two days of the current date, you can - and even should - convert them to UTC. 
+But as soon as you are handling datetimes that are more than a few days in advance (or back) you should **not** convert them to UTC but instead keep them in their local timezone. Why? You might want to read up on `why not to convert a datetime to a timestamp<https://andreas.heigl.org/2016/12/22/why-not-to-convert-a-datetime-to-timestamp/>`_
+
+Handling DateTimes with timezone-informations
+_____________________________________________
+
+Say we have an international calendaring application where users can add events that occur at different places worldwide and therefore at different timezones. To determine the exact time an event will happen means to save both the local time of the event and the timezone the event will happen in.
+
+Using the defalt DateTimeType will store the DateTime in the local time but without the timezone information. Therefore we need to store the timezone information as well and also need to provide a way to get the datetime back from the database with the correct timezone-information.
+
+To be able to transform these values back into their real timezone we have to save the timezone in a separate field of the entity requiring timezoned datetimes:
+
+.. code-block:: php
+
+    <?php
+    namespace Calendaring;
+
+    /**
+     * @Entity
+     */
+    class Event
+    {
+        /** @Column(type="datetime") */
+        private $eventDateTime;
+
+        /** @Column(type="string") */
+        private $timezone;
+
+        /**
+         * @var bool
+         */
+        private $localized = false;
+
+        public function __construct(\DateTimeInterface $eventDateTime)
+        {
+            $this->localized = true;
+            $this->eventDateTime = $eventDateTime;
+            $this->timezone = $$eventDateTime->getTimeZone()->getName();
+        }
+
+        public function getEventDateTime()
+        {
+            if (!$this->localized) {
+                
+                $this->eventDateTime = new DateTimeImmutable(
```

Comment:
This will trigger entity update as a side effect, that is not desired.

Reply:
How can that be avoided? I'm not too deep into that, so any help would be appreciated!

Reply:
What is the method supposed to do? Its name suggests just to return a datetime value but it changes `$eventDateTime` property.

Reply:
Added the reasoning in a code-comment.

I'd rather remove that but that would mean that there needs to be a way to modify that value correctly during hydration which currently doesn't work as I can only access the current value while converting from a DB value to a PHP value. But I would also need access to all the other values to be able to get the timezone string :frowning: